app/uk/gov/hmrc/servicecommissioningstatus/service/StatusCheckService.scala

```diff
@@ -0,0 +1,240 @@
+import uk.gov.hmrc.servicecommissioningstatus.connectors.{ArtifactoryConnector, FrontendRoute, GitHubConnector, Release, ReleasesConnector, ServiceConfigsConnector}
+import uk.gov.hmrc.servicecommissioningstatus.model.{AppConfigEnvironment, Dashboards, DeploymentEnvironment, Environment, FrontendRoutes, ServiceCommissioningStatus, StatusCheck}
+import uk.gov.hmrc.servicecommissioningstatus.model.Environment._
+
+import java.io.InputStream
+import java.util.zip.ZipInputStream
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+import scala.io.Source
+
+
+
+@Singleton
+class StatusCheckService @Inject()(
+  gitHubConnector         : GitHubConnector,
+  serviceConfigsConnector : ServiceConfigsConnector,
+  artifactoryConnector    : ArtifactoryConnector,
+  releasesConnector       : ReleasesConnector
+)(implicit ec: ExecutionContext){
+
+  def commissioningStatusChecks(serviceName: String)(implicit hc: HeaderCarrier): Future[ServiceCommissioningStatus] = {
+    for {
+      repo <- checkRepoExists(serviceName)
+
+      smConfig <- checkServiceManagerConfigExists(serviceName)
+
+      frontend <- isFrontend(serviceName)
+      routes   <- lookUpRoutes(serviceName)
+      intRoute  = checkFrontendRoute(frontend, routes, Integration)
+      devRoute  = checkFrontendRoute(frontend, routes, Development)
+      qaRoute   = checkFrontendRoute(frontend, routes, QA)
+      stagRoute = checkFrontendRoute(frontend, routes, Staging)
+      etRoute   = checkFrontendRoute(frontend, routes, ExternalTest)
+      prodRoute = checkFrontendRoute(frontend, routes, Production)
+
+      appConfigBase <- checkAppConfigBaseExists(serviceName)
+
+      appConfigInt  <- checkAppConfigExistsForEnv(serviceName, Integration)
+      appConfigDev  <- checkAppConfigExistsForEnv(serviceName, Development)
+      appConfigQA   <- checkAppConfigExistsForEnv(serviceName, QA)
+      appConfigStag <- checkAppConfigExistsForEnv(serviceName, Staging)
+      appConfigET   <- checkAppConfigExistsForEnv(serviceName, ExternalTest)
+      appConfigProd <- checkAppConfigExistsForEnv(serviceName, Production)
+
+      releases    <- releasesConnector.getReleases(serviceName).map(_.versions)
+      intDeployed  = checkIsDeployedForEnv(serviceName, releases, Integration)
+      devDeployed  = checkIsDeployedForEnv(serviceName, releases, Development)
+      qaDeployed   = checkIsDeployedForEnv(serviceName, releases, QA)
+      stagDeployed = checkIsDeployedForEnv(serviceName, releases, Staging)
+      etDeployed   = checkIsDeployedForEnv(serviceName, releases, ExternalTest)
+      prodDeployed = checkIsDeployedForEnv(serviceName, releases, Production)
+
+      kibana  <- checkKibanaDashboardExists(serviceName)
+      grafana <- checkGrafanaArchiveExists(serviceName)
+
+      buildJobs <- checkBuildJobsArchiveExists(serviceName)
+
+      sensuZip   <- artifactoryConnector.getSensuZip
+      alertConfig = checkAlertConfigExists(serviceName, sensuZip.get)
```

Comment:
You may want to consider collecting some of this in parallel. e.g. `(f1, f2).mapN((r1, r2) => ...` However we should keep the calls to the same service sequential.